app/uk/gov/hmrc/incometaxsessiondata/controllers/UserDetailsController.scala

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.incometaxsessiondata.controllers
+
+import play.api.libs.json.{JsError, JsSuccess, Json}
+import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
+import play.api.mvc.{Action, AnyContent, ControllerComponents}
+import uk.gov.hmrc.incometaxsessiondata.models.MtdItUser
+import uk.gov.hmrc.incometaxsessiondata.services.UserDetailsService
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton()
+class UserDetailsController @Inject()(cc: ControllerComponents,
```

Comment:
We are dealing with Session domain here, so user it not really applicable here;
I guess SessionController will make more sense here;
+  getById(sessionId: String) : ???
+  create(..)